src/SFA.DAS.AdminService.Web/Validators/Roatp/AllowedProviders/AddUkprnToAllowProvidersListValidator.cs

```diff
@@ -0,0 +1,46 @@
+﻿using FluentValidation;
+using SFA.DAS.AdminService.Web.Infrastructure.RoatpClients;
+using SFA.DAS.AdminService.Web.ViewModels.Roatp.AllowedProviders;
+
+namespace SFA.DAS.AdminService.Web.Validators.Roatp.AllowedProviders
+{
+    public class AddUkprnToAllowProvidersListValidator : AbstractValidator<AddUkprnToAllowedProvidersListViewModel>
+    {
+        private readonly IRoatpApplicationApiClient _applyApiClient;
+
+        public AddUkprnToAllowProvidersListValidator(IRoatpApplicationApiClient applyApiClient)
+        {
+            _applyApiClient = applyApiClient;
+
+            RuleFor(x => x.Ukprn).NotEmpty().WithMessage("Enter a UKPRN")
+                .DependentRules(() =>
+                {
+                    RuleFor(x => x.Ukprn).CustomAsync(async (ukprnInput, context, cancellationToken) =>
+                    {
+                        if (!int.TryParse(ukprnInput, out var ukprn))
+                        {
+                            context.AddFailure("Enter a valid UKPRN");
+                        }
+                        else if (ukprn < 10000000 || ukprn > 19999999)
+                        {
+                            context.AddFailure("Enter a valid UKPRN using 8 numbers");
+                        }
+                        else if (await _applyApiClient.GetAllowedProviderDetails(ukprn) != null)
```

Comment:
This code will run synchronously

Reply:
In the MVC pipeline it does but the only thing to do would be to use a GetAwaiter().GetResult() which I believe will be done under the hood.